### PR TITLE
multi_ec2 caching now correctly respects the --cache-only option

### PR DESCRIPTION
**Pull request by kwoodson**

### CODE REVIEW

**Comment by kwoodson** [comment]
@twiest please review

File: bin/openshift_ansible/awsutil.py [hunk 1]
```diff
@@ -31,10 +31,8 @@ def setup_host_type_alias_lookup(self):
             for value in values:
                 self.alias_lookup[value] = key
 
-
-
     def get_inventory(self,args=[]):
-        cmd = [self.inventory_path]
+        cmd = [self.inventory_path, '--cache-only']
```
Comment (twiest) [inline bin/openshift_ansible/awsutil.py]: Why do we want to force this?

File: inventory/multi_ec2.py [hunk 2]
```diff
@@ -59,20 +59,18 @@ def __init__(self):
         if self.args.cache_only:
             # get data from disk
             result = self.get_inventory_from_cache()
+            if result and self.is_cache_valid():
+                return
```
Comment (twiest) [inline inventory/multi_ec2.py]: What happens if we ask for cache-only and the cache isn't valid?

That doesn't seem to be handled here.

File: inventory/multi_ec2.py [hunk 3]
```diff
@@ -59,20 +59,18 @@ def __init__(self):
...
-        else:
-            # get data from disk
-            self.get_inventory_from_cache()
+            result = self.get_inventory()
+            return
+
+        # go fetch the inventories and cache them if cache is expired
+        self.get_inventory()
+        self.write_to_cache()
+        self.get_inventory()
```
Comment (twiest) [inline inventory/multi_ec2.py]: why are you calling get_inventory twice here?

**Pull request closed without merging**